okhttp/src/jvmMain/kotlin/okhttp3/OkHttpClient.kt

```diff
@@ -1004,6 +1005,21 @@ open class OkHttpClient internal constructor(
       callTimeout(duration.toMillis(), MILLISECONDS)
     }
 
+    /**
+     * Sets the default timeout for complete calls. A value of 0 means no timeout, otherwise values
+     * must be between 1 and [Integer.MAX_VALUE] when converted to milliseconds.
```

Comment:
Ick, converting 499 microseconds to no timeout is not good. You’re asking for a very small timeout and we’re giving you none at all.

Reply:
checkDuration gets it right. It rejects arguments smaller than 1ms.

https://github.com/square/okhttp/blob/ba3287b7486ec999d4a1960c8bd7eada85e0df3a/okhttp/src/jvmMain/kotlin/okhttp3/internal/-UtilJvm.kt#L100